plugin.go

```diff
@@ -0,0 +1,188 @@
+package sendremotefile
+
+import (
+	"errors"
+	"io"
+	"maps"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	rrErrors "github.com/roadrunner-server/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	RootPluginName    string        = "http"
+	PluginName        string        = "sendremotefile"
+	ContentTypeKey    string        = "Content-Type"
+	ContentTypeVal    string        = "application/octet-stream"
+	xSendRemoteHeader string        = "X-Sendremotefile"
+	defaultBufferSize int           = 10 * 1024 * 1024 // 10MB chunks
+	timeout           time.Duration = 5 * time.Second
+)
+
+type Configurer interface {
+	UnmarshalKey(name string, out any) error
+	Has(name string) bool
+}
+
+type Logger interface {
+	NamedLogger(name string) *zap.Logger
+}
+
+type Plugin struct {
+	log         *zap.Logger
+	writersPool sync.Pool
+}
+
+func (p *Plugin) Init(cfg Configurer, log Logger) error {
+	const op = rrErrors.Op("sendremotefile_plugin_init")
+
+	if !cfg.Has(RootPluginName) {
+		return rrErrors.E(op, rrErrors.Disabled)
+	}
+
+	p.log = log.NamedLogger(PluginName)
+
+	p.writersPool = sync.Pool{
+		New: func() any {
+			wr := new(writer)
+			wr.code = http.StatusOK
+			wr.data = make([]byte, 0, 10)
+			wr.hdrToSend = make(map[string][]string, 2)
+			return wr
+		},
+	}
+
+	return nil
+}
+
+func (p *Plugin) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		rrWriter := p.getWriter()
+		defer func() {
+			p.putWriter(rrWriter)
+			_ = r.Body.Close()
+		}()
+
+		next.ServeHTTP(rrWriter, r)
+
+		// if there is no X-Sendremotefile header from the PHP worker, just return
+		if url := rrWriter.Header().Get(xSendRemoteHeader); url == "" {
```

Comment:
After removing `reWriter`, you can simply check the response headers. So, no copy needed.